test/integration/targets/apt/tasks/apt.yml

```diff
@@ -140,6 +140,40 @@
 - name: uninstall hello with apt
   apt: pkg=hello state=absent purge=yes
 
+# verify that apt is handling multi-arch systems properly
+- name: add second architecture
+  command: dpkg --add-architecture {{ apt_foreign_arch }}
+
+- name: install hello:{{ apt_foreign_arch }} with apt
+  apt: pkg=hello:{{ apt_foreign_arch }} state=present update_cache=yes
+
+- name: uninstall hello:{{ apt_foreign_arch }} with apt
+  apt: pkg=hello:{{ apt_foreign_arch }} state=absent purge=yes
+
+- name: install deb file
+  apt: deb="/var/cache/apt/archives/hello_{{ hello_version.stdout }}_{{ apt_foreign_arch }}.deb"
+  register: apt_multi_initial
+
+- name: install deb file again
+  apt: deb="/var/cache/apt/archives/hello_{{ hello_version.stdout }}_{{ apt_foreign_arch }}.deb"
+  register: apt_multi_secondary
+
+- name: verify installation of hello:{{ apt_foreign_arch }}
+  assert:
+    that:
+        - "apt_multi_initial.changed"
+        - "not apt_multi_secondary.changed"
+
+- name: uninstall hello:{{ apt_foreign_arch }} with apt
+  apt: pkg=hello:{{ apt_foreign_arch }} state=absent purge=yes
+
+- name: uninstall all other {{ apt_foreign_arch }} packages that were installed as dependencies
+  apt: pkg={{ item }} state=absent purge=yes
+  with_lines: dpkg-query --showformat='${Package}:${Architecture}\n' --show '*:{{ apt_foreign_arch }}'
+
+- name: remove second architecture
```

Comment:
why? the tests are marked as destructive, and having the second arch enabled does not really hurt. on the other side blocks make playbooks less readable.